File: ruby/styles/document.rb
require_relative 'events'
require_relative 'navigation'

module Styles
  class Document < Base
    def prepare( compiler, article, parents)
      prev = article.has_any_content? ? article : nil
      article.children.each_index do |i|
        child = article.children[i]
        child.override_style( Navigation.new( child, prev, article.children[i+1]))
        prev  = child
      end
    end
    # def prepare( compiler, article, parents)
    #   article.children.each do |child|
    #     unless child.styled? || child.children?
    #       child.override_style( TextIndex.new)
    #     end
    #   end
    #   article
    # end

    def render( compiler, parents, article, data)
      data['index_style'] = 'document' if article.has_any_content?
      data['text_index']  = article.text_index( article.children)
    end
  end
end

File: java/src/com/alofmethbin/jadawin/styles/Document.java
package com.alofmethbin.jadawin.styles;

import com.alofmethbin.jadawin.Article;
import com.alofmethbin.jadawin.Page;

public class Document extends BaseStyle {

    @Override
    public String indexStyle(Article article) {
        return article.hasContent() ? "document" : super.indexStyle(article);
    }

    @Override
    public void prepare( Article article) {
        for (Page child: article.children()) {
            if ((! child.isStyled()) && (! child.hasChildren())) {
                child.setStyle( new TextIndex());
            }
        }
    }
}
